src/nbl/ext/ImGui/ImGui.cpp

```diff
@@ -304,287 +451,468 @@ namespace nbl::ext::imgui
+				assert(false);
+			}
 
 		smart_refctd_ptr<nbl::video::IGPUCommandBuffer> transistentCMD;
 		{
 			using pool_flags_t = IGPUCommandPool::CREATE_FLAGS;
-			// need to be individually resettable such that we can form a valid SIntendedSubmit out of the commandbuffer allocated from the pool
-			smart_refctd_ptr<nbl::video::IGPUCommandPool> pool = device->createCommandPool(families.id.transfer, pool_flags_t::RESET_COMMAND_BUFFER_BIT|pool_flags_t::TRANSIENT_BIT);
+
+			smart_refctd_ptr<nbl::video::IGPUCommandPool> pool = m_creationParams.utilities->getLogicalDevice()->createCommandPool(m_creationParams.transfer->getFamilyIndex(), pool_flags_t::RESET_COMMAND_BUFFER_BIT|pool_flags_t::TRANSIENT_BIT);
 			if (!pool)
 			{
-				logger->log("Could not create command pool!", system::ILogger::ELL_ERROR);
+				m_creationParams.utilities->getLogger()->log("Could not create command pool!", system::ILogger::ELL_ERROR);
 				assert(false);
 			}
 			
-			
 			if (!pool->createCommandBuffers(IGPUCommandPool::BUFFER_LEVEL::PRIMARY, 1u, &transistentCMD))
 			{
-				logger->log("Could not create transistent command buffer!", system::ILogger::ELL_ERROR);
+				m_creationParams.utilities->getLogger()->log("Could not create transistent command buffer!", system::ILogger::ELL_ERROR);
 				assert(false);
 			}
 		}
 
-		tQueue->startCapture();
-		{
-			// Setup Dear ImGui context
-			IMGUI_CHECKVERSION();
-			ImGui::CreateContext();
-
-			CreateFontSampler();
-			CreateDescriptorPool();
-			CreateDescriptorSetLayout();
-			CreatePipeline(renderpass, pipelineCache);
-			CreateFontTexture(transistentCMD.get(), tQueue);
-			prepareKeyMapForDesktop();
-			adjustGlobalFontScale();
-			UpdateDescriptorSets();
-		}
-		tQueue->endCapture();
+		// Dear ImGui context
+		IMGUI_CHECKVERSION();
+		ImGui::CreateContext();
 
-		auto & io = ImGui::GetIO();
-		io.DisplaySize = ImVec2(m_window->getWidth(), m_window->getHeight());
-		io.DisplayFramebufferScale = ImVec2(1.0f, 1.0f);
+		createPipeline();
+		createMDIBuffer();
+		createFontAtlasTexture(transistentCMD.get());
 
-		m_vertexBuffers.resize(maxFramesInFlight);
-		m_indexBuffers.resize(maxFramesInFlight);
+		auto & io = ImGui::GetIO();
+		io.BackendUsingLegacyKeyArrays = 0; // using AddKeyEvent() - it's new way of handling ImGUI events our backends supports
 	}
 
 	UI::~UI() = default;
 
-	void UI::createSystem()
+	void UI::createMDIBuffer()
 	{
-		system = system::IApplicationFramework::createSystem();
-		const auto logLevel = nbl::core::bitflag(system::ILogger::ELL_ALL);
-		logger = core::make_smart_refctd_ptr<system::CColoredStdoutLoggerWin32>(logLevel);
-		auto archive = core::make_smart_refctd_ptr<::ext::imgui::spirv::builtin::CArchive>(smart_refctd_ptr(logger));
-		
-		system->mount(core::smart_refctd_ptr(archive));
+		constexpr static uint32_t minStreamingBufferAllocationSize = 32u, maxStreamingBufferAllocationAlignment = 1024u * 64u, mdiBufferDefaultSize = /* 2MB */ 1024u * 1024u * 2u;
```

Comment:
eh, make it 128 min size and 4096 max alignment

Reply:
resolved